tools/tool-env/profile.ts

```diff
@@ -238,66 +296,326 @@ export function Profile<
   bucketName: string | ((...args: TArgs) => string),
   f: (...args: TArgs) => TResult | Promise<TResult>,
 ): typeof f {
-  if (! Profile.enabled) {
+  if (!Profile.enabled) {
     return f;
   }
 
   return Object.assign(function profileWrapper(this: any) {
-    const args = arguments as unknown as TArgs;
+    const args = Array.from(arguments) as TArgs;
 
     if (!running) {
       return f.apply(this, args);
     }
 
     const asyncLocalStorage = global.__METEOR_ASYNC_LOCAL_STORAGE;
-    let existingStore = asyncLocalStorage.getStore();
+    let store = asyncLocalStorage.getStore() || { currentEntry: [] };
+
+    const name = typeof bucketName === 'function' ? bucketName.apply(this, args) : bucketName;
+    
+    // callbacks with observer to track when the function finishes
+    const profileInfo = {
+      name,
+      isActive: false,
+      isCompleted: false,
+      startTime: Date.now()
+    };
+    
+    if (shouldRunInspectorProfiling(name)) {
+      profileInfo.isActive = startInspectorProfiling(name);
+      
+      if (profileInfo.isActive) {
+        // Monitor for process exit to stop profiling correctly
+        process.on('exit', () => {
+          if (profileInfo.isActive && !profileInfo.isCompleted) {
+            stopInspectorProfiling(name, true).catch(err => {
+              process.stdout.write(`[PROFILING_EXIT] Erro ao finalizar profiling: ${err}\n`);
+            });
+          }
+        });
+        
+        // Monitor for signals to stop profiling correctly
+        const signals = ['SIGINT', 'SIGTERM', 'SIGHUP'];
+        signals.forEach(signal => {
+          process.once(signal, () => {
+            if (profileInfo.isActive && !profileInfo.isCompleted) {
+              stopInspectorProfiling(name, true).catch(err => {
+                process.stdout.write(`[PROFILING_SIGNAL] Erro ao finalizar profiling: ${err}`);
```

Comment:
Eu consigo entender essa língua latina parecida, mas provavelmente usamos inglês aqui.

Reply:
lol, i forgot to remove this one, sorry